src/Build.OM.UnitTests/Construction/ProjectItemGroupElement_tests.cs

```diff
@@ -102,5 +104,136 @@ public void SetLabel()
             Assert.Equal("c", itemGroup.Label);
             Assert.True(project.HasUnsavedChanges);
         }
+
+        [Fact]
+        public void AddPackageReference_PreservesTrailingCommentOnSameLine()
+        {
+            string content =
+            """
+               <Project>
+                   <ItemGroup>
+                       <PackageReference Include="Newtonsoft.Json" Version="13.0.1" /><!-- some comment -->
+                   </ItemGroup>
+               </Project>
+            """;
+            using ProjectRootElementFromString projectRootElementFromString = new(content, ProjectCollection.GlobalProjectCollection, true);
+            ProjectRootElement project = projectRootElementFromString.Project;
+            ProjectItemGroupElement group = project.ItemGroups.First();
+
+            // Add a new PackageReference
+            group.AddItem("PackageReference", "Serilog").AddMetadata("Version", "4.3.0", expressAsAttribute: true);
+
+            string expectedContent =
+            """
+               <Project>
+                   <ItemGroup>
+                       <PackageReference Include="Newtonsoft.Json" Version="13.0.1" /><!-- some comment -->
+                       <PackageReference Include="Serilog" Version="4.3.0" />
+                   </ItemGroup>
+               </Project>
+            """;
+
+            Helpers.VerifyAssertLineByLine(expectedContent, project.RawXml);
+        }
+
+        private void AddItem_PreservesComments_Helper(string content, string expectedContent)
+        {
+            using ProjectRootElementFromString projectRootElementFromString = new(content, ProjectCollection.GlobalProjectCollection, true);
+            ProjectRootElement project = projectRootElementFromString.Project;
+            ProjectItemGroupElement group = project.ItemGroups.First();
+
+            // Insert a new item between the two existing items
+            var items = group.Items.ToList();
+            var newItem = project.CreateItemElement("PackageReference");
+            newItem.Include = "Inserted";
+            group.InsertAfterChild(newItem, items[0]);
+            newItem.AddMetadata("Version", "1.5.0", true);
+
+            Helpers.VerifyAssertLineByLine(expectedContent, project.RawXml);
+        }
+
+        [Fact]
+        public void AddItem_PreservesComments_VariousCases()
```

Comment:
Could you please consolidate both of the tests, so that you do not repeat the code? You can check various content using `Theory` instead of the `Fact` and passing different content to the test, like [here](https://github.com/dotnet/msbuild/blob/17babbf859e410fb09eaec1e84a0c6e375368641/src/Build.OM.UnitTests/Construction/ProjectItemElement_Tests.cs#L158).

Reply:
Updated